backend-agent/main.py

```diff
@@ -129,6 +137,58 @@ def check_health():
     return jsonify({'status': 'ok'})
 
 
+# Endpoint to fetch heatmap data from db
+@app.route('/api/heatmap', methods=['GET'])
+def get_heatmap():
+    try:
+        query = (
+            select(
+                ModelAttackScore.total_number_of_attack,
+                ModelAttackScore.total_success,
+                AttackModel.name.label("attack_model_name"),
+                Attack.name.label("attack_name"),
+                Attack.weight.label("attack_weight")
+            )
+            .join(AttackModel, ModelAttackScore.attack_model_id == AttackModel.id)
+            .join(Attack, ModelAttackScore.attack_id == Attack.id)
+        )
+
+        scores = db.session.execute(query).all()
+        all_models = {}
+        all_attacks = {}
+
+        for score in scores:
+            model_name = score.attack_model_name
+            attack_name = score.attack_name
+
+            if attack_name not in all_attacks:
+                all_attacks[attack_name] = score.attack_weight
+
+            if model_name not in all_models:
+                all_models[model_name] = {
+                    'name': model_name,
+                    'scores': {},
+                }
+
+            # Compute success ratio for this model/attack
+            success_ratio = (
+                round((score.total_success / score.total_number_of_attack) * 100)
+                if score.total_number_of_attack else 0
+            )
+
+            all_models[model_name]['scores'][attack_name] = success_ratio
+
+        return jsonify({
+            'models': list(all_models.values()),
+            'attacks': [
+                {'name': name, 'weight': weight}
+                for name, weight in sorted(all_attacks.items())
+            ]
+        })
+    except Exception as e:
```

Comment:
do you think we can restrict a bit this exception instead of using the generic Exception class? what can raise an exception (and which exception) in the try block?

Reply:
i will probably make an utility function for exception handling ? avoiding to avoid cluttering the code every time error handling is needed... need to check again